src/Tomcat/TomcatController.ts

```diff
@@ -445,15 +471,22 @@ export class TomcatController {
                 }
             });
 
+            const useStartupScripts: boolean = await Utility.getVSCodeConfigBoolean(Constants.CONF_USE_STARTUP_SCRIPTS);
             let startArguments: string[] = serverInfo.jvmOptions.slice();
-            if (serverInfo.getDebugPort()) {
-                startArguments = [`${Constants.DEBUG_ARGUMENT_KEY}${serverInfo.getDebugPort()}`].concat(startArguments);
+            let process: Promise<void>;
+            if (useStartupScripts) {
+                startArguments.unshift('run');
+                if (serverInfo.isDebugging()) {
+                    startArguments.unshift('jpda');
+                }
+            }
+            else {
+                startArguments.push('start');
             }
-            startArguments.push('start');
-            const javaProcess: Promise<void> = Utility.executeCMD(this._outputChannel, serverInfo.getName(), Utility.getJavaExecutable(), { shell: true }, ...startArguments);
+            process = Utility.executeCMD(this._outputChannel, serverInfo.getName(), Utility.getStartExecutable(serverInfo.getInstallPath()), { env: await this._tomcatModel.getEnvironmentVars(serverInfo, true), shell: true }, true, ...startArguments);
```

Comment:
Notable thing that changed: explicitly supplying the correct CATALINA_HOME, CATALINA_BASE, and JPDA_OPTS environment variables to the shell.